### DISCUSSION TITLE
How to add URI scheme to scope (SchemeNotInSetDecideRule)

### DISCUSSION THREAD

**Discussion by oschihin** [internetarchive/heritrix3#516] [Q&A]
In my crawls, I get several REJECTS in `SchemeNotInSetDecideRule` based on the data uri scheme (`data:`). How can I add this in the corresponding bean?

Thanks for a hint.

**Comment by ato** [comment]
I don't fully understand the question but Heritrix doesn't currently have a fetch module for `data:` URIs so I think adding them to scope would currently do nothing. I don't think queuing them makes sense anyway as the data is embedded within the URI itself. It would probably be best to treat them similarly to how style and script tags are treated and extract them immediately and never return them as "extracted" URIs at all.
